### DISCUSSION TITLE
how to transfer an initial field before solve ax=b

### DISCUSSION THREAD

**Discussion by ztdepztdep** [ginkgo-project/ginkgo#1135] [Q&A] [answered]
how to transfer an initial field before solve ax=b

**Comment by pratikvn** [comment, marked as answer]
Do you mean the initial solution, `x` ? If you have a pointer (say `dx`) to the data on `exec` which can be either `ReferenceExecutor` or `CudaExecutor`, then you can just do: 
(Updated):
```c++
auto x = gko::matrix::Dense<ValueType>::create(exec, 
                                               gko::dim<2>(num_elems, 1), 
                                               gko::array<ValueType>::view(exec, num_elems, dx),
                                               stride);
```

This will not allocate new memory, but use the data that `dx` points to and which is of size `num_elems`

**Reply by ztdepztdep** [reply]
/home/ztdep/lib/ginkgo/include/ginkgo/core/base/polymorphic_object.hpp(737): error: no instance of constructor "gko::matrix::Dense<ValueType>::Dense [with ValueType=double]" matches the argument list
            argument types are: (const std::shared_ptr<gko::Executor>, gko::dim<2UL, gko::size_type>, gko::array<double>)
          detected during instantiation of "std::unique_ptr<ConcreteType, std::default_delete<ConcreteType>> gko::EnableCreateMethod<ConcreteType>::create(Args &&...) [with ConcreteType=gko::matrix::Dense<double>, Args=<const std::shared_ptr<gko::Executor> &, gko::dim<2UL, gko::size_type>, gko::array<double>>]"

**Reply by upsj** [reply]
There is a parameter missing:
```cpp
auto x = gko::matrix::Dense<ValueType>::create(exec, 
                                               gko::dim<2>(num_elems, 1), 
                                               gko::array<ValueType>::view(exec, num_elems, dx),
                                               stride);
```
